tests/test_backends.py

```diff
@@ -0,0 +1,222 @@
+        result_df = result.df.compute()
+        assert_eq(result_df["cpu_lengths"], gt_lengths)
+
+    def test_cudf_backend(
+        self,
+        gpu_data,
+    ):
+        print("client", self.client)
+        dataset, gt_lengths = gpu_data
+        pipeline = GPUModule()
+        result = pipeline(dataset)
+        result_df = result.df.compute()
+        assert_eq(result_df["gpu_lengths"], gt_lengths)
+
+    def test_any_backend(
+        self,
+        cpu_data,
+        gpu_data,
+    ):
+        print("client", self.client)
+        cpu_dataset, gt_cpu_lengths = cpu_data
+        gt_cpu_lengths = gt_cpu_lengths.rename("any_lengths")
+        gpu_dataset, gt_gpu_lengths = gpu_data
+        gt_gpu_lengths = gt_gpu_lengths.rename("any_lengths")
+        pipeline = AnyModule()
+
+        cpu_result = pipeline(cpu_dataset)
+        cpu_result_df = cpu_result.df.compute()
+        assert_eq(cpu_result_df["any_lengths"], gt_cpu_lengths)
+        gpu_result = pipeline(gpu_dataset)
+        gpu_result_df = gpu_result.df.compute()
+        assert_eq(gpu_result_df["any_lengths"], gt_gpu_lengths)
+
+    def test_pandas_to_cudf(
+        self,
+        cpu_data,
+        gpu_data,
+    ):
+        print("client", self.client)
+        dataset, gt_cpu_lengths = cpu_data
+        _, gt_gpu_lengths = gpu_data
+        pipeline = Sequential(
+            [
+                CPUModule(),
+                ToBackend("cudf"),
+                GPUModule(),
+            ]
+        )
+        result = pipeline(dataset)
+        result_df = result.df.compute()
+        assert_eq(result_df["cpu_lengths"], gt_cpu_lengths)
+        assert_eq(result_df["gpu_lengths"], gt_gpu_lengths)
+
+    def test_cudf_to_pandas(
+        self,
+        cpu_data,
+        gpu_data,
+    ):
+        print("client", self.client)
+        _, gt_cpu_lengths = cpu_data
+        dataset, gt_gpu_lengths = gpu_data
+        pipeline = Sequential(
+            [
+                GPUModule(),
+                ToBackend("pandas"),
+                CPUModule(),
+            ]
+        )
+        result = pipeline(dataset)
+        result_df = result.df.compute()
+        assert_eq(result_df["cpu_lengths"], gt_cpu_lengths)
+        assert_eq(result_df["gpu_lengths"], gt_gpu_lengths)
+
+    def test_5x_switch(
+        self,
+        cpu_data,
+        gpu_data,
+    ):
+        print("client", self.client)
+        dataset, gt_cpu_lengths = cpu_data
+        _, gt_gpu_lengths = gpu_data
+        pipeline = Sequential(
+            [
+                CPUModule(),
+                ToBackend("cudf"),
+                GPUModule(),
+                ToBackend("pandas"),
+                CPUModule(),
+                ToBackend("cudf"),
+                GPUModule(),
+                ToBackend("pandas"),
+                CPUModule(),
+                ToBackend("cudf"),
+                GPUModule(),
+            ]
+        )
+        result = pipeline(dataset)
+        result_df = result.df.compute()
```

Comment:
add a small assert here for list(df.columns) which should be `id, text, gpu_lengths, cpu_lengths`